Turn on spell check for all TextEditors in Connect/Grammar/Diagnostic

cissyxyu opened the pull request:
## WHAT
Turn on spell check for the TextEditor component everywhere it appears in Connect, Diagnostic or Grammar admin panels (besides text boxes where students are typing their answer).

## WHY
So admin have an additional spell check step before they make edits to any student-facing text.

## HOW
Simply pass the shouldCheckSpelling flag to the React component when initialized.

### Screenshots
![Screenshot 2024-05-06 at 3 52 01 PM](https://github.com/empirical-org/Empirical-Core/assets/57366100/8d6b1bdf-31da-4dca-8882-612db1efc513)
![Screenshot 2024-05-06 at 3 51 50 PM](https://github.com/empirical-org/Empirical-Core/assets/57366100/959e676d-65ca-416d-ba6c-912238d7c8a7)
![Screenshot 2024-05-06 at 3 51 42 PM](https://github.com/empirical-org/Empirical-Core/assets/57366100/d93d3346-971d-4242-b436-f14ee2f283b6)
![Screenshot 2024-05-06 at 3 51 31 PM](https://github.com/empirical-org/Empirical-Core/assets/57366100/0a348b37-512f-4c53-aee6-e6ef2637847f)
![Screenshot 2024-05-06 at 3 51 22 PM](https://github.com/empirical-org/Empirical-Core/assets/57366100/1722958e-5ed7-4005-95c2-833725ea4df7)
![Screenshot 2024-05-06 at 3 51 10 PM](https://github.com/empirical-org/Empirical-Core/assets/57366100/93ca4ee0-ddb2-4fcd-99c2-1bd467e81e21)


### Notion Card Links
[(Please provide links to any relevant Notion card(s) relevant to this PR.)](https://www.notion.so/quill/Add-spell-check-to-the-Grammar-Diagnostic-and-Connect-CMSs-34aa46374067403184367163e906d3b6?pvs=4)

### What have you done to QA this feature?
Deploy to staging, go through the checklist on the Notion card and test all the flows listed, making sure that spell check is activated.

PR Checklist | Your Answer
------------ | -------------
Have you added and/or updated tests? |  manually tested
Have you deployed to Staging? | yes
Self-Review: Have you done an initial self-review of the code below on Github? | yes

cissyxyu committed:
all spelling check true added

cissyxyu committed:
all remaining text editors

emilia-friedberg reviewed (commented):
Are there any places this editor is used where we don't want spellcheck? I'm just wondering if it makes sense to just change the default behavior of the editor so we don't have to remember to apply this prop if we use it in another place.

brendanshean reviewed (commented):
It looks like some snapshot tests need to be updated

cissyxyu commented:
> Are there any places this editor is used where we don't want spellcheck? I'm just wondering if it makes sense to just change the default behavior of the editor so we don't have to remember to apply this prop if we use it in another place.

Yes, we don't want to use spell check in any boxes where the student is typing a response, because we want them to correct spelling in their response based on our feedback.

cissyxyu committed:
snapshot update

emilia-friedberg commented:
> Yes, we don't want to use spell check in any boxes where the student is typing a response, because we want them to correct spelling in their response based on our feedback.

@cissyxyu but isn't that a different text editor? I think we're just using `react-contenteditable` for those, not the shared draftjs editor you're passing this prop to now.

brendanshean commented:
I'm also curious about boilerplate Emilia mentioned.  Maybe even one step further with abstracting out: 

```
ContentState={ContentState} 
EditorState={EditorState}
```
as it seems to be in all of these examples.

cissyxyu committed:
change all instances

cissyxyu committed:
update snapshot tests

cissyxyu committed:
remove unused imports

cissyxyu committed:
diagnostic test

cissyxyu commented:
> > Yes, we don't want to use spell check in any boxes where the student is typing a response, because we want them to correct spelling in their response based on our feedback.
> 
> @cissyxyu but isn't that a different text editor? I think we're just using `react-contenteditable` for those, not the shared draftjs editor you're passing this prop to now.

I checked it out and you're right -- it looks like the TextEditor we use for student responses is actually a different component. I've updated the TextEditor component itself to have spellcheck, ContentState, and EditorState as inherent attributes rather than passed props, and changed all its references accordingly.

cissyxyu commented:
> I'm also curious about boilerplate Emilia mentioned. Maybe even one step further with abstracting out:
> 
> ```
> ContentState={ContentState} 
> EditorState={EditorState}
> ```
> 
> as it seems to be in all of these examples.

Good catch -- I've updated the component now to use ContentState, EditorState, and spellCheck inside the component rather than passed props! And I've removed those attributes from the outside references.

brendanshean reviewed (approved):
![Screenshot 2024-05-13 at 9 29 09 AM](https://github.com/empirical-org/Empirical-Core/assets/2057805/af3ecd8c-c7a9-4067-ac08-e0b12456175e) !!

Much better!  Nice work on this.
